packages/evm/src/types.ts

```diff
@@ -134,6 +136,26 @@ export type EVMEvents = {
   step: (data: InterpreterStep, resolve?: (result?: any) => void) => void
 }
 
+export interface EVMInterface {
+  journal: {
+    commit(): Promise<void>
+    revert(): Promise<void>
+    checkpoint(): Promise<void>
+    cleanJournal(): void
+    cleanup(): Promise<void>
+    putAccount(address: Address, account: Account): Promise<void>
+    deleteAccount(address: Address): Promise<void>
+    accessList?: Map<string, Set<string>>
+    addAlwaysWarmAddress(address: string, addToAccessList?: boolean): void
+    addAlwaysWarmSlot(address: string, slot: string, addToAccessList?: boolean): void
+    startReportingAccessList(): void
+  }
+  stateManager: EVMStateManagerInterface
+  precompiles: Map<string, PrecompileFunc>
+  runCall(opts: EVMRunCallOpts): Promise<EVMResult>
+  events?: AsyncEventEmitter<EVMEvents>
```

Comment:
Why are we making events optional? Could there be a valid EVM implementation that does not emit events?

Reply:
It is not super-necessary to make EVM run. If we make this non-optional we need to have "external" EVMs import our AsyncEventEmitter and implement the events it defines